core/server/common/src/main/java/alluxio/master/SafeModeManager.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+/**
+ * Interface for managing safe mode state.
+ */
+public interface SafeModeManager {
+
+  /**
+   * Enters safe mode.
+   */
+  void enterSafeMode();
```

Comment:
I am not sure if we always want only `SafeModeManager` to control whether we are in safe mode. User might also want to enter safe mode manually (It's not implemented yet but we might want to give user a command to do that in the future). Currently `DefaultSafeModeManager` is designed to be thread safe so it works if other master components want to put master in safe mode too.

Reply:
If we enter safe mode because the user specifically requested it, or because a block is under-replicated, we *dont* want to leave safe mode after `MASTER_WORKER_CONNECT_WAIT_TIME`. We should stay in safe mode until the user or block master tells us to exit. 

How about we add a parameter to `enterSafeMode` that says how long to stay in safe mode?Then we can add indefinite-time enter/exit safe mode methods later.

Reply:
I see. I agree we should have separate method for manually enter safe mode. Will rename the current method. We can add a method for manual safe mode later.